scripts/exgfs_aero_init_aerosol.py

```diff
@@ -0,0 +1,159 @@
+#! /usr/bin/env python3
+
+"""
+"""
+
+import os
+import subprocess
+import typing
+from datetime import datetime, timedelta
+from functools import partial
+
+# Constants
+atm_base_pattern = "{rot_dir}/{cdump}.%Y%m%d/%H/atmos/INPUT"           		  # Location of atmosphere ICs
+atm_file_pattern = "{path}/gfs_data.{tile}.nc"                                # Atm IC file names
+atm_ctrl_pattern = "{path}/gfs_ctrl.nc"                                       # Atm IC control file name
+restrt_base_pattern = "{rot_dir}/{cdump}.%Y%m%d/%H/atmos/RERUN_RESTART"       # Location of restart files (time of previous run)
+restrt_file_pattern = "{file_base}/{timestamp}fv_core.res.{tile}.nc"          # Name of restart data files (time when restart is valid)
+tracer_file_pattern = "{file_base}/{timestamp}fv_tracer.res.{tile}.nc"        # Name of restart tracer files (time when restart is valid)
+dycore_file_pattern = "{file_base}/{timestamp}fv_core.res.nc"                 # Name of restart dycore file (time when restart is valid)
+tracer_list_file_pattern = "{parm_gfs}/chem/gocart_tracer.list"               # Text list of tracer names to copy
+merge_script_pattern = "{ush_gfs}/merge_fv3_aerosol_tile.py"
+n_tiles = 6
+max_lookback = 4                                                              # Maximum number of past cycles to look for for tracer data
+debug = True
+
+# End configurable settings
+
+# Make sure print statements are flushed immediately, otherwise
+#   print statments may be out-of-order with subprocess output
+print = partial(print, flush=True)
+
+tiles = list(map(lambda t: "tile{t}".format(t=t), range(1, n_tiles + 1)))
+
+
+def main() -> None:
+	# Read in environment variables and make sure they exist
+	cdate = get_env_var("CDATE")
+	incr = int(get_env_var('STEP_GFS'))
+	fcst_length = int(get_env_var('FHMAX_GFS'))
+	cdump = get_env_var("CDUMP")
+	rot_dir = get_env_var("ROTDIR")
+	ics_dir = get_env_var("ICSDIR")
+	case = get_env_var("CASE")
+	ush_gfs = get_env_var("USHgfs")
+	parm_gfs = get_env_var("PARMgfs")
+	# data = get_env_var('DATA')
+
+	# os.chdir(data)
+
+	merge_script = merge_script_pattern.format(ush_gfs=ush_gfs)
+	tracer_list_file = tracer_list_file_pattern.format(parm_gfs=parm_gfs)
+
+	time = datetime.strptime(cdate, "%Y%m%d%H")
+	atm_source_path = time.strftime(atm_base_pattern.format(**locals()))
+
+	if(debug):
+		for var in ['merge_script', 'tracer_list_file', 'atm_source_path']:
+			print(f'{var} = {f"{var}"}')
+
+	atm_files, ctrl_files = get_atm_files(atm_source_path)
+	tracer_files, rest_files, core_files = get_restart_files(time, incr, max_lookback, fcst_length, rot_dir, cdump)
+
+	if (tracer_files is not None):
+		merge_tracers(merge_script, atm_files, tracer_files, rest_files, core_files[0], ctrl_files[0], tracer_list_file)
+
```

Comment:
merge_tracers function will merge files to the same file, this will cause rerunning issue, which the variables have already exist.